controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -331,6 +338,11 @@ protected void shutDown() throws Exception {
                 log.info("Awaiting termination of auto retention");
                 streamCutService.awaitTerminated();
             }
+
+            if (streamMetrics != null) {
```

Comment:
Does it make sense to put this in the `finally` block to make sure they're closed even in case of failure?

Reply:
Done.